tower-layer/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+//! Layer traits and extensions.
+//!
+//! A layer decorates an service and provides additional functionality. It
+//! allows other services to be composed with the service that implements layer.
+//!
+//! A middleware implements the [`Layer`] and [`Service`] trait.
+
+#![deny(missing_docs)]
+#![doc(html_root_url = "https://docs.rs/tower-layer/0.1.0")]
+
+extern crate futures;
+extern crate tower_service;
+
+#[cfg(feature = "util")]
+mod util;
+
+#[cfg(feature = "util")]
+pub use util::{Chain, LayerExt};
```

Comment:
Since we are bikeshedding, I'd probably avoid re-exporting `Chain`. I'd also make `mod util` public (and leave `Chain` in there).

docs.rs/tokio-buf